PhpAmqpLib/Helper/MiscHelper.php

```diff
@@ -67,23 +67,23 @@ public static function hexdump($data, $htmloutput = true, $uppercase = false, $r
         // Init
         $hexi = '';
         $ascii = '';
-        $dump = ($htmloutput === true) ? '<pre>' : '';
+        $dump = $htmloutput ? '<pre>' : '';
         $offset = 0;
         $len = mb_strlen($data, 'ASCII');
 
         // Upper or lower case hexidecimal
-        $x = ($uppercase === false) ? 'x' : 'X';
+        $hexFormat = $uppercase ? 'X' : 'x';
 
         // Iterate string
         for ($i = $j = 0; $i < $len; $i++) {
             // Convert to hexidecimal
-            $hexi .= sprintf("%02$x ", ord($data[$i]));
+            // We must use concatenation here because the $hexFormat value
+            // is needed for sprintf() to parse the format
+            $hexi .= sprintf('%02' .  $hexFormat . ' ', ord($data[$i]));
```

Comment:
because the variable contains the modifier type that is used for sprintf formating